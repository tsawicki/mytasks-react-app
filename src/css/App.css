@import url('https://fonts.googleapis.com/css?family=Lato:100,300,400');
@import url('https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css');

.App{
  text-align: center;
  max-width: 900px;
  margin-left: auto;
  margin-right: auto;
  padding: 15px;
}

.App h2{
  font-size: 6em;
  font-weight: 100;
  margin-top: 1em;
  margin-bottom: 0.5em; 
  color: #777;
}

@media (max-width: 767px){
  .App{
    padding: 5px;
  }
  .App h2{
    font-size: 3em;
  }
}