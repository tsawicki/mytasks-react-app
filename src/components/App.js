import React, { Component } from "react";
import NewTaskInput from "./NewTaskInput";
import TasksStats from "./TasksStats";
import TasksListHeader from "./TasksListHeader";
import TasksList from "./TasksList";

import "../css/App.css";

class App extends Component {
  constructor(props) {
    super(props);
    const storedTasks = this.readTasksFromLocalStorage();
    this.state = {
      tasks: storedTasks
    };
  }

  updateTasks = newTasks => {
    this.setState({
      tasks: newTasks
    });
  };

  addNewTask = newTask => {
    var newTasks = Array.from(this.state.tasks);
    newTasks.unshift(newTask);
    this.setState({
      tasks: newTasks
    });
  };

  readTasksFromLocalStorage() {
    return JSON.parse(localStorage.getItem("tasks") || "[]");
  }

  saveTasksToLocalStorage() {
    localStorage.setItem("tasks", JSON.stringify(this.state.tasks));
  }

  render() {
    this.saveTasksToLocalStorage();
    return (
      <div className="App">
        <h2>my tasks:</h2>
        <NewTaskInput onAddNewTask={this.addNewTask} />
        <TasksListHeader
          tasks={this.state.tasks}
          updateTasks={this.updateTasks}
        />
        <TasksList tasks={this.state.tasks} updateTasks={this.updateTasks} />
      </div>
    );
  }
}

export default App;
