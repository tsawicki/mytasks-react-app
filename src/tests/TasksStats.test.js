import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../components/App';

describe('TasksStats Component', () => {
  
  test('display stats', () => {
    const wrapper = mount(<App />);
    wrapper.setState({ tasks: [{"id":"9b122f8a-a3f6-4f02-8839-b54b46202acc","val":"trzy","isDone":false},{"id":"c3b324fd-e494-4912-81f2-ac122654e849","val":"dwa","isDone":true},{"id":"9217cd41-6f58-420e-8c63-22e2122f6877","val":"raz","isDone":true}] });
    const tasksStats = wrapper.find('.tasks-stats').text();
    expect(tasksStats).toMatchSnapshot();
  });

});

