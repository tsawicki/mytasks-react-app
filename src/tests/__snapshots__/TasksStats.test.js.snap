// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksStats Component display stats 1`] = `"3 TASKS:1 ACTIVE2 DONE"`;
